File: app/src/main/java/com/mytaxiapp/data/remote/VehicleInteractor.java
package com.mytaxiapp.data.remote;

import com.mytaxiapp.data.model.Vehicles;

import rx.Observable;

/**
 * @author Tosin Onikute.
 */

public interface VehicleInteractor {

    Observable<Vehicles> fetchVehicles(VehicleInterface vehicleInterface);

}

File: app/src/main/java/com/mytaxiapp/data/remote/VehicleInteractor.kt
package com.mytaxiapp.data.remote

import com.mytaxiapp.data.model.Vehicles

import rx.Observable

/**
 * @author Tosin Onikute.
 */

interface VehicleInteractor {

    fun fetchVehicles(vehicleInterface: VehicleInterface): Observable<Vehicles>

}
